### ISSUE TITLE
Install CapRover on a Private [Local] Network - CapRover Dashbaord NOT Accessible

### ISSUE THREAD

**Issue by abjoseph** [caprover/caprover#1206]
**IMPORTANT:** This is a bug report. If you are having problem with deploying a particular app use the deployment issue type.

> If you are having problems with deploying a particular app using a customized dockerfile - please change your questions to Dockerfile and post it where appropriate such as StackOverflow. This is a Docker question, not a CapRover specific issue.

--------------------------------------------------------------------------

What is the problem?
When trying to install CapRover on a Private [local] Network using the following [instructions](https://caprover.com/docs/run-locally.html#install-caprover-on-a-private-local-network), I'm never able to access port 3000 because the container that hosts the dashboard NEVER exposes/maps port 3000 to the host as shown below.

![image](https://user-images.githubusercontent.com/26844547/134787660-16c9d02b-fa95-44dc-a3ab-a4e0bf0df764.png)

As you can see, container *"captain-captain.1.wk6z1wopfn45ol12pbclz6744"* port 3000/tcp is internal only. So there is literally no way to access the CapRover dashboard from another PC on the network unless this is fixed. Unfortunately, it appears all the provisioning and configuration of those containers are baked into the caprover bootstrapping process and I'm not able to correct this oversight.

**Commands used to Install CapRover:**

``` shell
echo  "{\"skipVerifyingDomains\":\"true\"}" >  /captain/data/config-override.json

docker run -e MAIN_NODE_IP_ADDRESS=192.168.1.206 -p 80:80 -p 443:443 -p 3000:3000 -v /var/run/docker.sock:/var/run/docker.sock -v /captain:/captain caprover/caprover
```

If applicable, content of `captain-definition` file:

Steps to reproduce the problem:
Follow instructions @ [CapRover on a Private [local] Network](https://caprover.com/docs/run-locally.html#install-caprover-on-a-private-local-network).

Answers to the following questions where applicable:
- OS and Version: 
```
root@caprover:~# uname -a
Linux caprover 5.11.22-3-pve #1 SMP PVE 5.11.22-6 (Wed, 28 Jul 2021 10:51:12 +0200) x86_64 x86_64 x86_64 GNU/Linux
```
- RAM: 
```
root@caprover:~# free -h
              total        used        free      shared  buff/cache   available
Mem:           2.0G        284M        1.6G        264K        167M        1.7G
Swap:          256M          0B        256M
```

**Comment by githubsaturn** [comment]
Port `3000` is exposed through Docker's ingress network. You can't see it in `docker ps` but you should be able to see it in `docker service ls`. You can add an additional port by running
```
docker service update --publish-add published=3001,target=3000 captain-captain
```

You can read more about docker ingress network here: https://docs.docker.com/engine/swarm/ingress/

**Comment by githubsaturn** [comment]
Recently I'm seeing more of these issues. Are you running Ubuntu 20?

**Comment by githubsaturn** [comment]
I'm publishing a change that changes the `ingress` network to `host` - this should fix the issue for the OS versions that are having this problem.

Please wait about 40 minutes until the build is finished and try the edge version, and comment if you're still experiencing any issues.

**Comment by abjoseph** [comment]
Hi, thanks for the prompt reply. After you mentioned that it was using Docker Swarm and an ingress network. I did some more digging and discovered that LXC/LXD (which is what I was using on Proxmox) has known bugs and limitations with Overlay networking in Docker/DockerSwarm. Apparently this has been a long-standing issue and there were a slew of forum posts and blog articles that touched on the topic, suffice to say though there is no ** *proper* ** fix. 

So I have decided to switch to using a Full VM and the Ingress/Overlay network is working as expected on a single node. Thanks for your time and effort, I appreciate it. Hopefully, this helps the next person that comes across this issue if you're trying to use Linux (LXC/LXD) containers.
